File: code/null-conditional.cs
public class DataPoint
{
    public int Id { get; set; }
    public double Celsius { get; set; }
    public DataPoint? Child { get; set; }
}
        
var data = new DataPoint 
{ 
    Id = 1, 
    Celsius = 22.1, 
    Child = new DataPoint { Id = 2, Celsius = 22.8 }
};

var result = data.Child?.Child != null
                ? ToFahrenheit(data.Child.Child.Celsius) 
                : double.MinValue;

File: code/null-conditional.kt
data class DataPoint(val id: Int, val celsius: Double, 
    val child: DataPoint? = null)

val data = DataPoint(1, 22.1, DataPoint(2, 22.8))

val result = data.child?.child?.
            let { toFahrenheit(it.celsius) } ?: Double.MIN_VALUE